src/agg_bookend.c

```diff
@@ -0,0 +1,471 @@
+#include <postgres.h>
+#include <fmgr.h>
+#include <catalog/namespace.h>
+#include <nodes/value.h>
+#include <utils/lsyscache.h>
+#include <lib/stringinfo.h>
+#include <libpq/pqformat.h>
+
+/* bookend aggregates first and last:
+ *	 first(value, cmp) returns the value for the row with the smallest cmp element.
+ *	 last(value, cmp) returns the value for the row with the biggest cmp element.
+ *
+ * Usage:
+ *	 SELECT first(metric, time), last(metric, time) FROM metric GROUP BY hostname.
+ */
+
+extern Datum first_sfunc(PG_FUNCTION_ARGS);
```

Comment:
I believe these declarations are unnecessary as it is taken care of the PG_FUNCTION macros. No?